### ISSUE TITLE
There must be a way to supply `key` property to Snabbdom

### ISSUE THREAD

**Issue by gaydenko** [OutWatch/outwatch#29]
Hi! Let's look at the test case:
```Scala
import outwatch.dom._
import scala.scalajs.js.JSApp

object Test11 extends JSApp {
  val clicks = createHandler[Int](1)
  val nodes = clicks.map { i =>
    div(
      new Attribute(title = "key", value =s"key-$i"),
      a(href := "#", click(if (i == 1) 2 else 1) --> clicks,  s"I'm $i, go to another"),
      input()
    )
  }

  def main(): Unit = OutWatch.render("#app", div(child <-- nodes))
}
```
As you see, in accordance with Snabbdom logic `div` containing `key` attribute must be replaced as a child of containing `div` on clicking "..go to another". So it is expected these steps

- input something into the `input` field
- click href

will result in empty field, as far as previous `input` element must be completely removed form the DOM. Nevertheless the text which was input into the field is kept after clicking.

I have verified in browser `key` attribute value is changing on click.

**Comment by gaydenko** [comment]
I have looked at my old code where I played with Snabbdom few yeas ago. It seems like there are very different things: attributes and `key` property. That is attributes in dedicated object, whyle `key` is a "member field" or "property" of virtual node (a la `node.key`). Say, this is key place from Snabbdom:

```TypeScript
function sameVnode(vnode1: VNode, vnode2: VNode): boolean {
  return vnode1.key === vnode2.key && vnode1.sel === vnode2.sel;
}
```

I have looked at OutWatch code and at first glance haven't found properties in Snabbdom sense. @LukaJCB , have I missed them?

It would be nice if this finding is correct: it means there aren't bugs, but there is just a small must-have (not implemented yet) addition (well, small for those knowing OutWatch codebase, of course :) ).

**Comment by gaydenko** [comment]
And here how `key` is extracting from `data`:
```TypeScript
export function vnode(sel: string | undefined,
                      data: any | undefined,
                      children: Array<VNode | string> | undefined,
                      text: string | undefined,
                      elm: Element | Text | undefined): VNode {
  let key = data === undefined ? undefined : data.key;
  return {sel: sel, data: data, children: children,
          text: text, elm: elm, key: key};
}
```

**Comment by LukaJCB** [comment]
Awesome, thanks for writing this up! :)

**Comment by LukaJCB** [comment]
Fixed by 31afe5c0e2e49ff9227dec4a1217807702edb68c

**Comment by gaydenko** [comment]
@LukaJCB Thanks for the fix! - you are  fast 👍 
0.9.4?
